app/questionnaire/rules/operations.py

```diff
@@ -214,3 +229,24 @@ def evaluate_map(
             )
 
         return results
+
+    def evaluate_option_label_from_value(self, value: str, answer_id: str) -> str:
+        label: str = ""
+        answers = self.schema.get_answers_by_answer_id(answer_id)
+        label_options: str = next(
+            (
+                options["label"]
+                for answer in answers
+                for options in answer["options"]
+                if value == options["value"]
+            ),
+            "",
+        )
```

Comment:
Personally, I think a list/dict comphrenshion would be more readable than a generator with `next`. However, not a blocker. Performance diff is negligible.

Reply:
the intention was to assign default `str` if `value` is not found , not the performance.  In list comprehension , you have to convert `list` to str and i don't think you can assign default using list/dict comprehension.

Reply:
Value should never not be found if it is a valid schema. Only reason it might not be found is for dynamic options which we won't support here anyway.

> In list comprehension , you have to convert list to str

Not sure I follow. You are already creating a generator, a list comp would be similar. i.e.

```py
        label_options: str = [
            options["label"]
            for answer in answers
            for options in answer["options"]
            if value == options["value"]
        ][0]
```

or:

```py
        label_options: str = {
            options["label"]: options["value"]
            for answer in answers
            for options in answer["options"]
        }[value]
```

Reply:
In my implementation, it assigns empty string `""` to `label_options`  if  generator raises `StopIteration` error , guarding against any error. 
Your suggestion may need another `if` condition in order to assign empty string `label_option = ""`  when `value` and `option['label']` don't match.
Happy to change if you think list/dict comp makes more sense

Reply:
I don't have anything against the gen, I just think it is clearer with list/dict comp personally. In both cases we would get an error anyway, list would get index error and dict would get key error if value didn't exist, so that is covered, and we don't need to assign the label to an empty str as we should expect to find a value.

Reply:
I have seen instances where we return empty string as default if condition fails hence i used generator to assign empty. i agree list/dict comp is more readable. will update it to list comp